src/main/java/data/AnalysisResults.java

```diff
@@ -67,13 +75,23 @@ public Builder setStore(String store) {
       return this;
     }
 
+    public Builder setTimestamp(long timestamp) {
+      this.timestamp = Optional.of(Long.valueOf(timestamp));
+      return this;
+    }
+
     public AnalysisResults build() {
       // No logo was detected.
       if (store == null) {
         store = Optional.empty();
       }
 
-      return new AnalysisResults(rawText, categories, store);
+      // No date was detected.
+      if (timestamp == null) {
+        timestamp = Optional.empty();
+      }
```

Comment:
This check could be avoided by initializing `timestamp` to `Optional.empty()`. I did this for `store` in the updated master branch.